optuna/pruners/_hyperband.py

```diff
@@ -1,9 +1,9 @@
+from __future__ import annotations
+
 import binascii
 import math
 from typing import Container
-from typing import List
 from typing import Optional
```

Comment:
Could you use `ANOTER_TYPE | None` instead of `optinal[ANOTHER_TYPE]` to address the original issue?
https://peps.python.org/pep-0604/#proposal

```suggestion
```